plugins/rest/rest_test.go

```diff
@@ -1216,3 +1226,26 @@ func getTestServerWithTimeout(d time.Duration) (baseURL string, teardownFn func(
 	})
 	return ts.URL, ts.Close
 }
+
+func mockAuthPluginLookup(name string) HTTPAuthPlugin {
+	if name == "my_plugin" {
+		return &myPluginMock{}
+	}
+	return nil
+}
+
+type myPluginMock struct{}
+
+func (m *myPluginMock) NewClient(c Config) (*http.Client, error) {
+	tlsConfig, err := DefaultTLSConfig(c)
+	if err != nil {
+		return nil, err
+	}
+	return DefaultRoundTripperClient(
+		tlsConfig,
+		defaultResponseHeaderTimeoutSeconds,
+	), nil
+}
+func (m *myPluginMock) Prepare(req *http.Request) error {
```

Comment:
[nit]
```suggestion
func (*myPluginMock) Prepare(*http.Request) error {
```

but this really really really doesn't need changing. 😄

Reply:
all nits are appreciated! 😁